src/OrangePaymentSlipData.php

```diff
@@ -220,17 +253,26 @@ public function getCompleteReferenceNumber($formatted = true, $fillZeros = true)
             $completeReferenceNumber = str_pad($referenceNumber, 26, 'X', STR_PAD_LEFT);
         } elseif ($this->getWithBankingCustomerId()) {
             // Get reference number and fill with zeros
-            $referenceNumber = str_pad($referenceNumber, 20, '0', STR_PAD_LEFT);
+            $padLength = 26 - $this->bankingCustomerIdLength;
+            $referenceNumber = str_pad($referenceNumber, $padLength, '0', STR_PAD_LEFT);
             // Prepend banking customer identification code
             $completeReferenceNumber = $this->getBankingCustomerId() . $referenceNumber;
+            
+            if (strlen($completeReferenceNumber) !== 26) {
+                throw new PaymentSlipException('Reference number must have 26 digits');
+            }
+            
+            if (!is_numeric($completeReferenceNumber)) {
+                throw new PaymentSlipException('Reference number must be a numerical value');
```

Comment:
Often, I echo out the variable which lead to the exception, e.g. ``Reference number "%s" must be a numerical value``